15-greenhouse-monitor/README.md

```diff
@@ -0,0 +1,369 @@
+            "new_value": 20
+        }
+    }
+}
+```
+
+## Alerts
+
+In addition to reporting the sensor values, the app can send an alert should a particular sensor value be too high or too low. The app uses environment variables to configure alert ranges for each sensor. The alert ranges are:
+
+* `normal`: Describes the normal range of values for the sensor. The app sends a `warning` alert when the sensor value is higher than the `high` threshold or lower than the `low` threshold.
+
+* `warning`: Describes the range of values for which the app sends a `warning` alert. When defined, values falling outside the `warning` range produce a `critical` alert. Values inside the `warning` range but outside the `normal` range produce a `warning` alert.
+
+You configure the ranges using environment variables that include the sensor name and range to set, following this format:
+
+  * `<sensor-name>_<range>_low`: Configures the lower bound of a range for the named sensor.
+  * `<sensor-name>_<range>_high`: Configures the upper bound of the range for the named sensor.
+
+> **Note**: You don't have to configure both `high` an `low` values for a range - the application checks just the thresholds provided.
+
+Some examples:
+
+* `light_level_normal_low=100`: sets the lower bound of normal readings from the `light_level` sensor to 100. A `warning` alert is produced when the `light_level` value is below 100.
+
+* `soil_temp_normal_high=40`, `soil_temp_warning_high=50`: sets the upper bound for normal and warning ranges for the `soil_temp` sensor. With this configuration, alerts are produced as follows:
+
+  * No alert is produced when the soil temperature is below 40.
+
+  * A warning alert is produced when the soil temperature is between 40 and 50.
+
+  * A critical alert is produced when the soil temperature is above 50.
+
+
+### Alert Events
+
+When the app detects an alert condition, it captures details of the alert and sends an event to `alert.qo`. The event includes the overall alert level (warning/critical) and details of the sensor readings, similar to this:
+
+```json
+{
+    "alert": "warning",
+    "alert_seq": "first",
+    "app": "nf15",
+    "air_humidity": {
+        "status": "ok",
+        "value": 34.740234375
+    },
+    "air_pressure": {
+        "status": "ok",
+        "value": 1016.686328125
+    },
+    "air_temp": {
+        "status": "ok",
+        "value": 23.46999931335449
+    },
+    "light_level": {
+        "alert": "warning",
+        "status": "high",
+        "value": 279
+    },
+    "soil_moisture": {
+        "status": "ok",
+        "value": 342
+    },
+    "soil_temp": {
+        "status": "ok",
+        "value": 26.38705444335938
+    }
+}
+```
+
+The event has these properties:
+
+* `alert`: Describes the overall alert level. This is the combined alert level of all of the sensors. Values are `warning`, `critical` or not present, meaning no alert, which happens when sensor readings return to normal.
```

Comment:
> This is the combined alert level of all of the sensors.

Is it the highest level present between all the alerts? If so, I would state that explicitly.